diff --git a/node_modules/expo-modules-core/ios/Legacy/EXUtilities.m b/node_modules/expo-modules-core/ios/Legacy/EXUtilities.m
index d4cd1df..e93cd61 100644
--- a/node_modules/expo-modules-core/ios/Legacy/EXUtilities.m
+++ b/node_modules/expo-modules-core/ios/Legacy/EXUtilities.m
@@ -46,7 +46,16 @@ - (UIViewController *)currentViewController
     return [utilService currentViewController];
   }
 
-  UIViewController *controller = [[[UIApplication sharedApplication] keyWindow] rootViewController];
+  UIViewController *controller = /* Runtime patched */ UIViewController *controller = nil;
+    if ([[UIApplication class] respondsToSelector:@selector(sharedApplication)]) {
+        id sharedApp = [[UIApplication class] performSelector:@selector(sharedApplication)];
+        if (sharedApp && [sharedApp respondsToSelector:@selector(keyWindow)]) {
+            id keyWindow = [sharedApp performSelector:@selector(keyWindow)];
+             if (keyWindow && [keyWindow respondsToSelector:@selector(rootViewController)]) {
+                 controller = [keyWindow performSelector:@selector(rootViewController)];
+             }
+        }
+    };
   UIViewController *presentedController = controller.presentedViewController;
 
   while (presentedController && ![presentedController isBeingDismissed]) {
@@ -234,7 +243,11 @@ + (BOOL)catchException:(void(^)(void))tryBlock error:(__autoreleasing NSError **
   if ([[[[NSBundle mainBundle] bundlePath] pathExtension] isEqualToString:@"appex"]) {
     return nil;
   }
+  #if defined(RCT_APP_EXTENSION)
+  return nil;
+#else
   return [[UIApplication class] performSelector:@selector(sharedApplication)];
+#endif
 }
 
 NSError *EXErrorWithMessage(NSString *message)
diff --git a/node_modules/expo-modules-core/ios/ReactDelegates/ExpoReactDelegate.swift b/node_modules/expo-modules-core/ios/ReactDelegates/ExpoReactDelegate.swift
index 1e36208..fd0f6bf 100644
--- a/node_modules/expo-modules-core/ios/ReactDelegates/ExpoReactDelegate.swift
+++ b/node_modules/expo-modules-core/ios/ReactDelegates/ExpoReactDelegate.swift
@@ -23,8 +23,13 @@ public class ExpoReactDelegate: NSObject {
       .compactMap { $0.createReactRootView(reactDelegate: self, moduleName: moduleName, initialProperties: initialProperties, launchOptions: launchOptions) }
       .first(where: { _ in true })
       ?? {
-        guard let appDelegate = (UIApplication.shared.delegate as? (any ReactNativeFactoryProvider)) ??
-          ((UIApplication.shared.delegate as? NSObject)?.value(forKey: "_expoAppDelegate") as? (any ReactNativeFactoryProvider)) else {
+        var sharedApp: UIApplication?
+        let selector = NSSelectorFromString("sharedApplication")
+        if UIApplication.responds(to: selector) {
+            sharedApp = UIApplication.perform(selector)?.takeUnretainedValue() as? UIApplication
+        }
+        guard let appDelegate = (sharedApp?.delegate as? (any ReactNativeFactoryProvider)) ??
+          ((sharedApp?.delegate as? NSObject)?.value(forKey: "_expoAppDelegate") as? (any ReactNativeFactoryProvider)) else {
           return reactNativeFactory.recreateRootView(withBundleURL: nil, moduleName: moduleName, initialProps: initialProperties, launchOptions: launchOptions)
         }
 
